include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,465 @@
+            _program =  "ftp -Vo";
+#endif
+        else
+            _program.clear();
+    }
+#endif  // defined(STDLIB_VS)
+
+    void _updateCommand()
+    {
+        if (!_program.empty())
+        {
+            _command = _program + " " + _path + "/" + _name + ".version " + _url;
+#if defined(STDLIB_VS)
+            _command = _command + "; exit  [int] -not $?}\" > nul 2>&1";
+#else
+            _command = _command + " > /dev/null 2>&1";
+#endif
+        }
+    }
+
+    inline void operator()(std::promise<bool> versionCheckProm)
+    {
+        _checkForNewerVersion(*this, std::move(versionCheckProm));
+    }
+};
+
+// ==========================================================================
+// Metafunctions
+// ==========================================================================
+
+// ==========================================================================
+// Functions
+// ==========================================================================
+
+// ----------------------------------------------------------------------------
+// Function setURL()
+// ----------------------------------------------------------------------------
+
+inline void setURL(VersionCheck & me, std::string url)
+{
+    std::swap(me._url, url);
+    me._updateCommand();
+}
+
+// ----------------------------------------------------------------------------
+// Function _getOS()
+// ----------------------------------------------------------------------------
+
+constexpr const char * _getOS()
+{
+    //get system information
+#ifdef __linux
+    return "Linux";
+#elif __APPLE__
+    return "MacOS";
+#elif defined(STDLIB_VS)
+    return "Windows";
+#elif __FreeBSD__
+    return "FreeBSD";
+#elif __OpenBSD__
+    return "OpenBSD";
+#else
+    return "unknown";
+#endif
+}
+
+// ----------------------------------------------------------------------------
+// Function _getBitSys()
+// ----------------------------------------------------------------------------
+
+constexpr const char * _getBitSys()
+{
+#if SEQAN_IS_32_BIT
+    return "_32_";
+#else
+     return "_64_";
+#endif
+}
+
+// ----------------------------------------------------------------------------
+// Function _checkWritability()
+// ----------------------------------------------------------------------------
+
+#if defined(STDLIB_VS)
+inline bool _checkWritability(std::string const & path)
+{
+    DWORD ftyp = GetFileAttributesA(path.c_str());
+    if (ftyp == INVALID_FILE_ATTRIBUTES || !(ftyp & FILE_ATTRIBUTE_DIRECTORY))
+    {
+        if (!CreateDirectory(path.c_str(), NULL))
+            return false;
+    }
+
+    HANDLE dummyFile; // check writablity by trying to create a file in GENERIC_WRITE mode
+    std::string fileName(path + "/dummy.txt");
+    dummyFile = CreateFile(fileName.c_str(), GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+
+    if (dummyFile == INVALID_HANDLE_VALUE)
+        return false;
+
+    CloseHandle(dummyFile);
+    DeleteFile(fileName.c_str());
```

Comment:
Please add an assertion that the return value must be non-zero. And return the with false if ```DeleteFile``` returns ```0```